c10/core/InferenceMode.h

```diff
@@ -36,9 +40,17 @@ struct TORCH_API InferenceMode {
   //    }
   //    `k.add_(2)` still need to go through InplaceOrView kernel so that it's
   //    prepared for future autograd.
+  //
+  // 3. Why does setting InferenceMode also set GradMode?
+  //
+  //    This is required since InferenceMode is a faster and more restricive
+  //    version of NoGradMode. All runtime checks using GradMode::is_enabled()
```

Comment:
Instead of `NoGradMode` I think it should probably say `GradMode(false)`... I'd say it's worth being fussy about the names here given how many similar ones are flying around: `GradMode`, `AutoGradMode`, `NoGradGuard`, ... there's no actual `NoGradMode`, is there? 😬 

In that same vein it might be clearer in the "[Expected TLS state in InferenceMode]" note above to say "InferenceMode(true)" and "InferenceMode(false)", or "InferenceMode enabled/disabled" instead of "InferenceMode" vs "NormalMode" - unless you're using an idiom that I'm just unfamiliar with, in which case ofc disregard 😁